crates/bevy_ui/src/ui_node.rs

```diff
@@ -112,8 +112,51 @@ pub enum Val {
     VMax(f32),
 }
 
+impl PartialEq for Val {
```

Comment:
This can be a separate discussion though, it doesn't need to block this PR as your implementation is definitely an improvement over the `PartialEq` derive.

Reply:
`Val`s definitely can't be ordered, but checking for equality like this seems sound, even if you're right that the evaluated result is much more often what you need. 

I think that implementing the `PartialEq` trait is useful in general for external consumers: it's often a critical foundation of tests and debugging.

Reply:
Oh that's true, I hadn't considered debugging.

Reply:
I do agree that it's a very contextual unit to try and do equality with, but I do think it brings more utility than possible confusion. The fact that `PartialOrd` isn't implemented should indicate pretty solidly to consumers that they need to be careful with comparisons.